src/main/java/spoon/support/visitor/ClassTypingContext.java

```diff
@@ -643,7 +646,7 @@ private boolean isSubTypeArg(CtTypeReference<?> subArg, CtTypeReference<?> super
 		if (typeRef instanceof CtTypeParameterReference) {
 			CtTypeParameterReference typeParamRef = (CtTypeParameterReference) typeRef;
 			CtTypeParameter typeParam = typeParamRef.getDeclaration();
-			if (typeParam.getTypeParameterDeclarer() instanceof CtExecutable) {
+			if (typeParam != null && typeParam.getTypeParameterDeclarer() instanceof CtExecutable) {
 				//the parameter is declared in scope of Method or Constructor
```

Comment:
So it sounds like another bug of `CtTypeParameterReference#getDeclaration()`
And then I do not like the idea to ignore null here. It should throw SpoonExpeception instead.